src/components/views/Registration.js

```diff
@@ -48,15 +48,34 @@ const Registration = () => {
   const [username, setUsername] = useState("");
   const [name, setName] = useState("");
   const [password, setPassword] = useState("");
-  let errorResponse = null;
   const [birthday, setBirthday] = useState(null);
   const [gender, setGender] = useState(null)
+  const [err, setErr] = useState("")
+
+  // This little function asks the server if the typed in username is available and sets an error message accordingly
+  useEffect(() => {
+    async function checkAvailability(){
+      try{
+        const requestBody = JSON.stringify({username})
+        const response = await api.post('/users/usernames', requestBody)
+        // We don't want to show an error if the user retypes his old username, only if it is different and taken
```

Comment:
In the registration, shouldn't it not be allowed to type your "old" username, as you are still creating a new entity with the same username

Reply:
Ah yes, sorry, removed the misleading comment. It isn't allowed.